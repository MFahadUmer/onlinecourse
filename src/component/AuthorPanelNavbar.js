import React from 'react';
import { Link } from 'react-router-dom';

const AuthorPanelNavbar = () => (
  <div>
    <h2>Welcome To Author Panel</h2>
    <Link to="/addCourse">Add Course</Link>
    <Link to="/authorCourses">Add Course</Link>
    <Link to="/logout">Logout</Link>
  </div>
);

export default AuthorPanelNavbar;
