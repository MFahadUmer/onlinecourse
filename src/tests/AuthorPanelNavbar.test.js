import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import AuthorPanelNavbar from '../component/AuthorPanelNavbar';

describe('Author Panel Navbar Test', () => {
  it('should render author panel navbar', () => {
    render(
      <BrowserRouter>
        <AuthorPanelNavbar />
      </BrowserRouter>,
    );
    expect(screen.getByText('Add Course')).toBeTruthy();
  });

  it('should render author panel navbar', () => {
    render(
      <BrowserRouter>
        <AuthorPanelNavbar />
      </BrowserRouter>,
    );
    expect(screen.getByText('Add Course').href).toBe('http://localhost/addCourse');
  });

  it('should render author panel navbar', () => {
    render(
      <BrowserRouter>
        <AuthorPanelNavbar />
      </BrowserRouter>,
    );
    expect(screen.getByText('Logout').href).toBe('https://reactonlinecourse.herokuapp.com/');
  });

  it('matches the Navbar component snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <AuthorPanelNavbar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
