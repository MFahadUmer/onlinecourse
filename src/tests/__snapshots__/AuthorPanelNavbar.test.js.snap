// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Panel Navbar Test matches the Navbar component snapshot 1`] = `
<div
  className="authorNavbar"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <span
      className="fas fa-arrow-circle-left authorNavbarLink"
    />
  </a>
  <a
    className="authorNavbarLink"
    href="/addCourse"
    onClick={[Function]}
  >
    Add Course
  </a>
  <a
    className="authorNavbarLink"
    href="https://reactonlinecourse.herokuapp.com/"
  >
    Logout
  </a>
</div>
`;
